contrib/php/template.py

```diff
@@ -0,0 +1,276 @@
+        "ext/soap/tests/bug73037.phpt",
+        "ext/soap/tests/server009.phpt",
+        "ext/sockets/tests/bug63000.phpt",
+        "sapi/fpm/tests/socket-ipv4-fallback.phpt",
+        # Marked as XFAIL on PHP repo in March 2024
+        "ext/openssl/tests/openssl_error_string_basic_openssl3.phpt",
+        "ext/openssl/tests/openssl_private_decrypt_basic.phpt",
+        # Broken and being discussed upstream
+        # https://github.com/php/php-src/issues/11252
+        "ext/gd/tests/bug43073.phpt",
+        "ext/gd/tests/bug48732-mb.phpt",
+        "ext/gd/tests/bug48732.phpt",
+        "ext/gd/tests/bug48801-mb.phpt",
+        "ext/gd/tests/bug48801.phpt",
+        "ext/gd/tests/bug53504.phpt",
+        "ext/gd/tests/bug65148.phpt",
+        "ext/gd/tests/bug73272.phpt",
+        # https://github.com/php/php-src/issues/13806
+        "ext/openssl/tests/openssl_x509_parse_basic.phpt",
+        # Under investigation
+        "ext/gettext/tests/bug53251.phpt",
+        "ext/gettext/tests/gettext_bind_textdomain_codeset-retval.phpt",
+        "ext/gettext/tests/gettext_bindtextdomain-cwd.phpt",
+        "ext/posix/tests/posix_errno_variation1.phpt",
+        "ext/zip/tests/oo_encryption.phpt",
+        "sapi/cli/tests/009.phpt",
+        "sapi/cli/tests/012-2.phpt",
+        "sapi/fpm/tests/bug77780-header-sent-error.phpt",
+    ]:
+        self.rm(f)
+
+
+def pre_install(self):
+    self.make_install_args += [
+        f"INSTALL_ROOT={self.chroot_destdir}",
+    ]
+
+
+def post_install(self):
+    self.install_license("LICENSE")
+    self.install_file("README.md", "usr/share/doc/php")
+    self.install_file(
+        self.files_path / "sysusers.conf",
+        "usr/lib/sysusers.d",
+        name="php-fpm.conf",
+    )
+    self.install_service(self.files_path / "php-fpm")
+    # default php-fpm config files
+    self.install_file(self.files_path / "php-fpm.conf", "etc/php")
+    self.install_file(self.files_path / "www.conf", "etc/php/php-fpm.d")
+    # these are unnecessary with apk backups
+    self.rm(self.destdir / "etc/php/*.default", glob=True)
+    self.rm(self.destdir / "etc/php/php-fpm.d/*.default", glob=True)
+    # extensions
+    extcp = self.destdir / "etc/php/conf.d"
+    self.mkdir(extcp, parents=True)
+    for f in (self.destdir / "usr/lib/php/20230831/").glob("*.so"):
+        extso = f.name.split("/")[-1]
+        extn = extso.removesuffix(".so")
+        with open(extcp / f"{extn}.ini", "w") as outf:
+            if extn == "opcache":
+                outf.write(f"zend_extension={extso}\n")
+            else:
+                outf.write(f"extension={extso}\n")
+
+
+def _extension(extn, iif):
+    @subpackage(f"{pkgname}-{extn}")
+    def _ext(self):
+        self.pkgdesc = f"{pkgdesc} ({extn} extension)"
+
+        if iif:
+            self.install_if = [f"{pkgname}={pkgver}-r{pkgrel}"]
+
+        return [
+            f"etc/php/conf.d/{extn}.ini",
+            f"?usr/include/php/ext/{extn}",
```

Comment:
do we want to ship include files with the modules? this should probably remain in devel unless you have a good reason